src/AcmTable/AcmTable.tsx

```diff
@@ -358,22 +364,41 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
     }, [sorted, page, perPage])
 
     const rows = useMemo<IRow[] | undefined>(() => {
-        const newRows = paged.map((item) => {
+        const newRows: IRow[] = []
+        paged.forEach((item, i) => {
             const key = keyFn(item)
-            return {
+            let isOpen: boolean | undefined = undefined
+            if (props.subItems?.length && props.groupFn) {
+                const group = props.groupFn(item, props.subItems ?? [])
+                if (group.length) {
+                    isOpen = expanded[key] ?? false
+                }
+            }
+            newRows.push({
+                isOpen,
                 selected: selected[key] === true,
                 props: { key },
                 cells: columns.map((column) => {
-                    return typeof column.cell === 'string' ? (
-                        get(item as Record<string, unknown>, column.cell)
-                    ) : (
-                        <Fragment key={key}>{column.cell(item)}</Fragment>
-                    )
+                    return typeof column.cell === 'string'
+                        ? get(item as Record<string, unknown>, column.cell)
+                        : { title: <Fragment key={key}>{column.cell(item)}</Fragment> }
                 }),
+            })
+            if (isOpen !== undefined && props.groupFn) {
+                const group = props.groupFn(item, props.subItems || [])
+                newRows.push({
+                    parent: i,
+                    // fullWidth: true,
+                    cells: [
+                        {
+                            title: props.subItemCellTransform?.(group),
+                        },
+                    ],
```

Comment:
Maybe just require the cluster UI use cases to return the object with title key; so that app-ui can return a multi-index array